### DISCUSSION TITLE
installing bundle using streaming giving error : BLK_DEV_NBD not enabled in kernel

### DISCUSSION THREAD

**Discussion by abdo12007** [rauc/meta-rauc-community#51] [Announcements]
i have bitbaked a core-image-base for the rspberry pi 
i have made sure that this options kernel are enable in the rauc.cfg : 

CONFIG_BLK_DEV_LOOP=y
CONFIG_SQUASHFS=y
CONFIG_MD=y
CONFIG_BLK_DEV_DM=y
CONFIG_DM_VERITY=y
CONFIG_CRYPTO_SHA256=y

then i generate the img , flashing it on an sd card and it works on the raspberry pi3
then i generate a bundle in the plain format and i installed it using rauc install ... and it works fine.

then

uncommenting this option on the .bb file 
RAUC_BUNDLE_FORMAT = "verity"

and i generated another bundle in the verity format and i installed it using streaming 
but i get this error :  
failed mounting bundle: failed to resolve 'ndb' netlink family - BLK_DEV_NBD not enabled in kernel 

why i got  BLK_DEV_NBD not enabled in kernel although i have enable it on the rauc.cfg ? 
could that be because i am using an sd card ?
how can i solve this ?

**Comment by leon-anavi** [comment]
Hi @abdo12007,

Which Yocto release and branch for meta-rauc-raspberrypi are you using?

Best regards, Leon

**Comment by abdo12007** [comment]
hello @leon-anavi 
i am using Yocto Dunfell
meta-rauc-raspberrypi = "dunfell:738db79bd714f7ca5969482a2c7cab80471d8e9f

**Comment by leon-anavi** [comment]
Hi @abdo12007, 

Support for streaming installation from a HTTP(S) server for bundles in verity was added in RAUC version [v1.7](https://github.com/rauc/rauc/releases/tag/v1.7) and I see meta-rauc dunfell as of the moment provide v1.8 so in this aspect you should be ok.

Have you added `CONFIG_BLK_DEV_NBD=y` to [the rauc.cfg kernel fragment for Raspberry Pi](https://github.com/rauc/meta-rauc-community/blob/master/meta-rauc-raspberrypi/recipes-kernel/linux/files/rauc.cfg)? 

Best regards,
Leon

**Comment by abdo12007** [comment]
yes i have added CONFIG_BLK_DEV_NBD=y .
this is the content of $PWD/poky-rpi-rauc/meta-rauc/recipes-kernel/linux/linux-yocto/rauc.cfg 
CONFIG_BLK_DEV_LOOP=y
CONFIG_SQUASHFS=y
CONFIG_MD=y
CONFIG_BLK_DEV_DM=y
CONFIG_DM_VERITY=y
CONFIG_CRYPTO_SHA256=y
CONFIG_BLK_DEV_NBD=y

always the same issue

**Comment by abdo12007** [comment]
hello @leon-anavi 
after i rebuild the hole image 
now i am getting this error : 
Failed mounting bundle: failed to open /dev/mapper/control: no such file or directory

**Comment by ejoerns** [comment]
@abdo12007 please check if the kernel generated actually has DM enabled. It does not look like if this would be the case.

In your kernel build dir you should see the actual `.config` generated. If this looks fine, please check if the kernel built is actually the kernel installed on the target (e.g. by comparing checksums).

**Comment by abdo12007** [comment]
hello, 
I verified and the DM is enable go this error : 
failed mounting bundle: failed to load dm table: invalid argument 
are you familiar with this error ?

**Comment by ejoerns** [comment]
> hello, I verified and the DM is enable go this error : failed mounting bundle: failed to load dm table: invalid argument are you familiar with this error ?

This could mean you have enabled DM support but not DM_VERITY.

**Comment by abdo12007** [comment]


**Reply by ejoerns** [reply]
It is a kernel option (`CONFIG_DM_VERITY`) and you already mentioned it in the above config fragment.

Make sure that you apply the config fragment to the kernel you are using (above one assumes you are using `linux-yocto`).

Please refer to general yocto project documentation on yocto usage and how to manipulate kernel config since this is out-of-scope here.
